mimic/test/test_identity_osksadm.py

```diff
@@ -0,0 +1,366 @@
+"""
+Tests for mimic identity :mod:`mimic.rest.identity_api`
+"""
+
+from __future__ import absolute_import, division, unicode_literals
+
+import json
+import uuid
+
+import ddt
+from six import text_type
+
+from twisted.trial.unittest import SynchronousTestCase
+from twisted.internet.task import Clock
+
+from mimic.core import MimicCore
+from mimic.resource import MimicRoot
+from mimic.test.dummy import make_example_external_api, ExternalApiStore
+from mimic.test.helpers import json_request, request
+
+
+@ddt.ddt
+class TestIdentityMimicOSKSCatalogAdminListExternalServices(SynchronousTestCase):
+    """
+    Tests for ``/identity/v2.0/services``, provided by
+    :obj:`mimic.rest.idenity_api.IdentityApi`
+    """
+    def setUp(self):
+        self.core = MimicCore(Clock(), [])
+        self.root = MimicRoot(self.core).app.resource()
+        self.uri = "/identity/v2.0/services"
+        self.eeapi_name = u"externalServiceName"
+        self.headers = {
+            b'X-Auth-Token': [b'ABCDEF987654321']
+        }
+        self.verb = b"GET"
+
+    def test_auth_fail(self):
+        """
+        GET with no X-Auth-Token header results in 401.
+        """
+        (response, json_body) = self.successResultOf(
+            json_request(self, self.root, self.verb,
+                         self.uri))
+
+        self.assertEqual(response.code, 401)
+        self.assertEqual(json_body['unauthorized']['code'], 401)
+
+    @ddt.data(
+        0, 1, 10
+    )
+    def test_listing(self, api_entry_count):
+        """
+        GET will list the registered services.
+        """
+        api_list = []
+
+        # create the desired number of services per test parameter
+        for _ in range(api_entry_count):
```

Comment:
Loops in tests are a code smell, and comprehensions can be nicer (when they work) for building up lists than manually calling `.append`.  Consider `api_list = [ExternalApiStore(...) for ignored in range(api_entry_count)]` instead.